core/common/src/Clock.kt

```diff
@@ -53,5 +53,19 @@ private class InstantTimeMark(private val instant: Instant, private val clock: C
     override fun minus(duration: Duration): TimeMark = InstantTimeMark(instant - duration, clock)
 }
 
+/**
+ * Returns the [Clock] by storing an initial [TimeMark] using [TimeSource.markNow] and [returns][Clock.now] the elapsed
+ * time using [TimeMark.elapsedNow] plus the provided [offset].
```

Comment:
```suggestion
 * Creates a [Clock] that uses the [time mark at the moment of creation][TimeMark.markNow] to determine how [far][TimeMark.elapsedNow]
 * the [current moment][Clock.now] is from the [origin].
```
In my opinion, this conveys the "what", whereas the current version is focused on the "how".